client/state/atomic-transfer/use-atomic-transfer-query.ts

```diff
@@ -0,0 +1,70 @@
+import { useQuery } from '@tanstack/react-query';
+import wpcom from 'calypso/lib/wp';
+import { transferStates, TransferStates } from 'calypso/state/automated-transfer/constants';
+import { SiteSlug } from 'calypso/types';
+
+interface ErrorResponse {
+	code: 'no_transfer_record';
+}
+
+interface SuccessResponse {
+	status: TransferStates;
+}
+
+const fetchLatestAtomicTransfer = (
+	siteSlug: SiteSlug
+): Promise< ErrorResponse | SuccessResponse > =>
+	wpcom.req.get( {
+		path: `/sites/${ siteSlug }/atomic/transfers/latest`,
+		apiNamespace: 'wpcom/v2',
+	} );
+
+interface UseAtomicTransferQueryOptions {
+	refetchInterval?: number;
+}
+
+const endStates: TransferStates[] = [
+	transferStates.NONE,
+	transferStates.COMPLETE,
+	transferStates.COMPLETED,
+	transferStates.FAILURE,
+	transferStates.ERROR,
+	transferStates.REVERTED,
+];
+
+export function useAtomicTransferQueryQueryKey( siteSlug ) {
+	return [ 'sites', siteSlug, 'atomic', 'transfers', 'latest' ];
+}
+
+export const useAtomicTransferQuery = (
```

Comment:
I wonder if we need to define this hook since several mechanisms already exist:
- [`useAtomicTransfer`](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/checkout/checkout-thank-you/marketplace/use-atomic-transfer.tsx#L16) hook
- [`WithAtomicTransfer` HOC](https://github.com/Automattic/wp-calypso/blob/trunk/client/sites-dashboard/components/with-atomic-transfer.tsx#L40) (has a different hook inside)
- [an automated transfer handler](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/data-layer/wpcom/sites/automated-transfer/status/index.js#L47) (this is what `hosting/main.js` is using currently)

Seems like we could also unify these solutions.

Reply:
We can surely tidy these up. I think `use-atomic-transfer.tsx` could be simplified by using the same hook we're adding here.

I think unification can be done as a separate PR as it will only delay this longer. Daniel posted another PR we can consider for unification, too.

Reply:
We could revisit this as a follow-up.